File: src/main/java/com/example/demo/service/RecipeService.java
package com.example.demo.service;

import java.util.List;

import com.example.demo.dto.RecipeDto;

public interface RecipeService {
    RecipeDto getRecipeById(Integer recipeId);
    void saveRecipe(RecipeDto recipeDto);
    List<RecipeDto> getAllRecipes();
}

File: bin/src/main/resources/static/js/app/RecipeService.js
'use strict'

angular.module('demo.services', []).factory('RecipeService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getRecipeById = function(recipeId) {
				var url = CONSTANTS.getRecipeByIdUrl + recipeId;
				return $http.get(url);
			}
			service.getAllRecipes = function() {
				return $http.get(CONSTANTS.getAllRecipes);
			}
			service.saveRecipe = function(recipeDto) {
				return $http.post(CONSTANTS.saveRecipe, recipeDto);
			}
			return service;
		} ]);